### PR TITLE
Deprecate nGraph API to don't include it to public API

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - Deprecate get_version
 - Deprecate functions from `util.hpp`
 - Removed `slice_plan` from the top level

### Tickets:
 - *ticket-id*

### CODE REVIEW

**Comment by ilyachur** [comment]
@GlebKazantaev I moved `slice_plan` from top level to `op/itils`. Do you agree with this place? We use this class in our transformations and some ops.
Should I also change the namespace to `ngraph::op::utils`?

**Review by mbencer — APPROVED** [review]

**Review by tsocha — APPROVED** [review]

File: ngraph/core/src/op/transpose.cpp [hunk 1]
```diff
@@ -42,6 +42,7 @@ void op::v1::Transpose::validate_and_infer_types() {
 
     set_input_is_relevant_to_shape(1);
 
+    NGRAPH_SUPPRESS_DEPRECATED_START
```
Comment (GlebKazantaev) [inline ngraph/core/src/op/transpose.cpp:45]: Why we add deprecations to the methods which are used in existing opset operations? Do we have a replacement for them?

Comment (ilyachur) [reply ngraph/core/src/op/transpose.cpp:45]: We will remove it from only public API, just move to private part

**Comment by ilyachur** [comment]
@GlebKazantaev Please have a look

**Pull request merged**